### PR TITLE
sys/crypto: Add PSA Crypto wrappers

### PR DESCRIPTION
**Pull request by Einhornhool**
### Contribution description
This adds wrappers for the PSA Crypto module to the RIOT cipher module

### Issues/PRs references
Prerequisite: #18578

Split up PR #18547

### CODE REVIEW

File: makefiles/pseudomodules.inc.mk [hunk 1]
```diff
@@ -191,6 +191,7 @@ PSEUDOMODULES += posix_headers
 PSEUDOMODULES += printf_float
 PSEUDOMODULES += prng
 PSEUDOMODULES += prng_%
+PSEUDOMODULES += psa_riot_cipher_%
```
Comment (leandrolanzieri) [inline makefiles/pseudomodules.inc.mk:194]: It's preferred to explicitly list all modules when possible.

File: sys/crypto/psa_riot_cipher/Kconfig [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config MODULE_PSA_RIOT_CIPHER_AES_128_ECB
+    bool
```
Comment (leandrolanzieri) [inline sys/crypto/psa_riot_cipher/Kconfig:9]: Is it intended that all these modules are not selectable by the user?

File: sys/crypto/psa_riot_cipher/aes_128_cbc.c [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    status = psa_cipher_generate_iv(&operation, output, operation.default_iv_length, &iv_length);
+    if (status != PSA_SUCCESS) {
+        return status;
+    }
+
+    status = cbc_aes_common(&operation.backend_ctx.cipher_ctx.aes_128, key_buffer, key_buffer_size,
+                            output, input, input_length, output + operation.default_iv_length,
+                            output_length);
+
+    (void)output_size;
```
Comment (leandrolanzieri) [inline sys/crypto/psa_riot_cipher/aes_128_cbc.c:56]: Shouldn't the output size be checked to see if there's enough space?

File: sys/crypto/psa_riot_cipher/Kconfig (hunk 2 above)
Comment (Einhornhool) [reply sys/crypto/psa_riot_cipher/Kconfig:9]: Yes, the user chooses the RIOT backend in `sys/psa_crypto/Kconfig` (as seen on #18578) and the rest is selected automatically.

File: makefiles/pseudomodules.inc.mk (hunk 1 above)
Comment (Einhornhool) [reply makefiles/pseudomodules.inc.mk]: Done

File: sys/crypto/psa_riot_cipher/aes_128_cbc.c (hunk 3 above)
Comment (Einhornhool) [reply sys/crypto/psa_riot_cipher/aes_128_cbc.c]: Done

File: sys/crypto/psa_riot_cipher/Kconfig (hunk 2 above)
Comment (leandrolanzieri) [reply sys/crypto/psa_riot_cipher/Kconfig:9]: Ok then!

**Review by leandrolanzieri — COMMENTED** [review]
Some more comments

**Comment by Einhornhool** [comment]
To make this easier to test and work on are going to keep working on the original branch: https://github.com/RIOT-OS/RIOT/pull/18547

**Pull request closed without merging**